topics/03_file_handling_and_regex/HW_03_file_handling_and_regex.py

```diff
@@ -0,0 +1,164 @@
+file_path = 'test_data.txt'
+line_count = 0
+word_count = 0
+with open(file_path, 'r') as file:
+    for line in file:
+        line_count += 1
+        words = line.split()
+        word_count += len(words)
+print(f"Total lines: {line_count}")
+print(f"Total words: {word_count}")
+# 6 Write a script that replaces a specific word (e.g., "error") in a file logs.txt with the word "fixed" and saves the changes.
+with open('logs.txt', 'r') as file:
+    data = file.readlines()
+
+with open('logs.txt', 'w') as file:
+    for line in data:
+        new_line = line.replace('error', 'fixed')
+        file.write(new_line)
+# 7 Write a script that tries to read a file data.txt, handles the exception if the file does not exist, and prompts the user to create the file if it's missing.
+def read_file():
+    try:
+        with open('data.txt', 'r') as file:
+            data = file.read()
+            print("File content:")
+            print(data)
+    except FileNotFoundError:
+        print("The file 'data.txt' does not exist.")
+        create_file = input("Would you like to create the file? (yes/no): ").strip().lower()
+        if create_file == 'yes':
+            with open('data.txt', 'w') as file:
+                file.write("This is a new file.")
+            print("'data.txt' has been created.")
+        else:
+            print("No file was created.")
+
+# 8 Merging multiple files: Write a script that reads multiple files (file1.txt, file2.txt, file3.txt), merges their content, and writes the result to a new file called merged.txt.
+file_paths = ['file1.txt', 'file2.txt','file3.txt']
+
+with open('merged.txt', 'w', encoding="utf-8") as output_file:
+    for file_path in file_paths:
+        with open(file_path, 'r', encoding="utf-8") as input_file:
+            for line in input_file:
+                output_file.write(line)
+# 9 Create a script that reads a configuration file (e.g., config.ini) and prints out the key-value pairs. Use the configparser library to handle the file.
+import configparser
+config = configparser.ConfigParser()
+config.read(file_name)
+for section in config.sections():
+    print(f"[{section}]")
+    for key, value in config.items(section):
+        print(f"{key} = {value}")
+    print()
+ # 10 Write a script that reads a log file (e.g., server.log) and counts how many times a specific error message appears. Print the number of occurrences of that error.
+def count_error_messages(logfile_path, error_prefix):
+    # Initialize a counter to zero
+    error_count = 0
+
+    # Open and read the log file
+    try:
+        with open(logfile_path, 'r') as file:
+            for line in file:
+                # Check if the error prefix exists in the current line
+                if error_prefix in line:
+                    error_count += 1
+    except FileNotFoundError:
+        print(f"Error: The file '{logfile_path}' does not exist.")
+        return
+    except Exception as e:

```

Comment:
Handling general Exception is a bad practice, common practice is handling Exceptions that you think might happen, such as FileNotFoundException, that you used